provisioner/k8s/README.md

```diff
@@ -1,3 +1,112 @@
-# k8s
+# k8s Provisioner
 
-A concrete Provisioner implementation for "plain" k8s bundles.
+## Summary
+
+The k8s provisioner is a core rukpak provisioner that knows how to interact with `k8s` format bundles. 
+The `k8s` bundles, or plain bundles, are simply container images containing a set of static kubernetes YAML manifests in a given directory.  
+For example, below is an example Dockerfile that builds a plain `k8s` bundle from a directory containing static manifests.  
+```dockerfile
+FROM scratch
+LABEL manifests="/manifests"
+COPY ./manifests /manifests
+```
+
+The `k8s` provisioner is able to unpack a given `k8s` bundle onto a cluster and then instantiate it, making the
+content of the bundle available in the cluster. It does so by reconciling the `Bundle` and `BundleInstance` types that 
+have the `spec.provisionerClassName` field set to `core.rukpak.io/plain`. This field must be set to the correct provisioner name 
+in order for the k8s provisioner to see and interact with the bundle. 
+
+## Use cases
+
+### Install and apply a specific version of a bundle 
+
+The k8s provisioner can install and make available a specific `k8s` bundle available in the cluster. 
+
+Simply create a `Bundle` resource pointing to a specific version of your bundle, and a `BundleInstance` which references that bundle.
+The k8s provisioner will unpack the provided Bundle onto the cluster, and eventually make the content available on the cluster. 
+
+```yaml
+apiVersion: core.rukpak.io/v1alpha1
+kind: Bundle
+metadata:
+  name: my-bundle
+spec:
+  image: my-bundle-image@sha256:xyz123
+  provisionerClassName: core.rukpak.io/plain
+---
+apiVersion: core.rukpak.io/v1alpha1
+kind: BundleInstance
+metadata:
+  name: my-bundle-instance
+spec:
+  provisionerClassName: core.rukpak.io/plain
+  bundleName: my-bundle
+``` 
+### Make bundle content available but do not install it
+
+There is a natural separation between sourcing of the content and application of that content via two separate RukPak APIs, `Bundle` and `BundleInstance`. 
+A user can specify a particular `Bundle` to be available in the cluster for inspection before any application of the resources. 
+Given a `Bundle` resource named `my-bundle`, the k8s provisioner will pull down and unpack the bundle to a variety of storage backends, such as ConfigMaps or PersistentVolumes.
+
+Currently the ConfigMap backend is supported by the provisioner, and the content can be inspected by viewing the metadata ConfigMap, which holds
+information on all ConfigMaps related to this bundle. 
+
+`kubectl -n rukpak-system get cm --selector core.rukpak.io/owner-name=<bundle-name>` gets all ConfigMaps related to the `Bundle` on-cluster.
```

Comment:
Can we make a note that the `rukpak-system` is the default namespace for deploying provisioner-related system resources? I've been playing around with making this namespace configurable in #72.

Reply:
Also: any value in outputting a sample Bundle metadata ConfigMap?

Reply:
sure, that sounds good 👍

Reply:
Also, this and similar lines may need to change based on my comment here: https://github.com/operator-framework/rukpak/pull/70/files#r816268935

Reply:
> make a note
I assume that our manifests will use the `rukpak-system` namespace? If so, maybe it isn't necessary in the context of the readme to get into the detail about the namespace being configurable?

Reply:
Yeah that's a good point. I still think it's worth noting the `rukpak-system` namespace is the default namespace before we start documenting kubectl commands. None of my comments were blocking though.

Reply:
I kept the mention of the default namespace but did not mention overriding it since our manifests are hardcoded. 

Also decided not to output the metadata configmap as that feels pretty low-level and not part of a top-level README necessarily.